# Taxonomy of Biological Pathways

Static layouts vs automated
Comparisons of different species


## Researchers at UIC

- [Dr. Qun-Tian Wang](qtwang@uic.edu)
  - Proteomics, epigenetic maintenance of adult heart function
- Dr. Nava Segev
  - Intracellular Trafficking
  - Probably not really applicable?
- [Dr. Nissim Hay](nhay@uic.edu)
- [Dr. Pradip Raychaudhuri](pradip@uic.edu)
  - DNA damage response, tumorigenesis, metastasis
  - Directed me to Damiano Fantini
- Dr. Damiano Fantini
  - High throughput gene expression analysis
  - Postdoctoral Research Associate

## Researchers at Arizona

- Dr. Guang Yao
- Dr. Ryan Gutenkunst

## Researchers Francesco Interviewed

- Federica Zill
- Ankit Jambusaria

## Papers
### Taxonomy Papers
- [Wehrend and Lewis](http://dl.acm.org/citation.cfm?id=949531.949553)
- [Pretorius et al](http://dx.doi.org/10.1007/978-3-319-06793-3_5)
- [Ahn et al](http://dx.doi.org/10.1109/TVCG.2013.238)

### Other
- [Omics Vis Survey](http://www.nature.com/nmeth/journal/v7/n3s/full/nmeth.1436.html)
- [Weighted graph comparison techniques for brain connectivity analysis](http://dl.acm.org/citation.cfm?doid=2470654.2470724)

## Edited notes from Fintan, September 29 2015

> It really does seem to me that there is enough new information and good research there for it to be accepted, it is the structure and organization that is causing it problems.

> The task list is very good and I think is probably the strongest contribution in the paper, but I believe the reviewers are correct in saying that it is not a taxonomy yet.

> I think that you need to read some other taxonomies and structure yours similarly, and cite them in you paper. (Wehrend and Lewis)[http://dl.acm.org/citation.cfm?id=949531.949553] is one of the earliest attempts at this sort of thing for general visualization.

> I like (Pretorius et al)[http://dx.doi.org/10.1007/978-3-319-06793-3_5]'s
recent approach for describing tasks for multivariate graph vis.
It might help structure yours.

> (Ahn et al)[http://dx.doi.org/10.1109/TVCG.2013.238]'s task taxonomy
for dynamic graphs is also worth having a look at for how he breaks
up his categorization.

> I do have concerns about the review of visualization techniques.
It reads more like a survey of biological visualization tools.  
There have been many full surveys of bio vis tools in the past,
often in high impact journals e.g. (here)[http://www.nature.com/nmeth/journal/v7/n3s/full/nmeth.1436.html]
It may be possible to do a more up to date  review,
but such a STAR report would end up being
much longer than your current paper is.

> The point is that having a small review doesn't add much to this paper,
and there are more in depth reviews available.

> I understand that you might have wanted to describe visualization techniques
that relate to specific items in you task list your task list,
but that is not clear.

> I think that a proper multi-layer categorization of your task list,
and then giving examples of visualization techniques
(rather than general applications)
that are applicable for different categories of task
and their combinations would definitively improve the strength of the paper.

> I think that by doing this you will also have an opportunity to identify more
gaps in existing visualization techniques for addressing what you have
identified as important tasks
(kind of like you have done with the section on visualizing uncertainty).
You may also have a chance to suggest existing techniques as solutions
that do not exist in current biological applications
(for example, you gave matrices very little detail,
but they may be superior to node link graphs for some tasks
e.g. (here)[http://dl.acm.org/citation.cfm?doid=2470654.2470724] )
