% ---------------------------------------------------------------------------
% Author guideline and sample document for EG publication using LaTeX2e input
% D.Fellner, v1.13, Jul 31, 2008

\documentclass{egpubl}
\usepackage{eurovis2015}

% --- for  Annual CONFERENCE
\ConferenceSubmission   % uncomment for Conference submission
% \ConferencePaper        % uncomment for (final) Conference Paper
% \STAR                   % uncomment for STAR contribution
% \Tutorial               % uncomment for Tutorial contribution
% \ShortPresentation      % uncomment for (final) Short Conference Presentation
% \Areas                  % uncomment for Areas contribution
% \MedicalPrize           % uncomment for Medical Prize contribution
% \Education              % uncomment for Education contribution
%
% --- for  CGF Journal
% \JournalSubmission    % uncomment for submission to Computer Graphics Forum
% \JournalPaper         % uncomment for final version of Journal Paperâˆ«
%
% --- for  CGF Journal: special issue
% \SpecialIssueSubmission    % uncomment for submission to Computer Graphics Forum, special issue
% \SpecialIssuePaper         % uncomment for final version of Journal Paper, special issue
%
% --- for  EG Workshop Proceedings
% \WsSubmission    % uncomment for submission to EG Workshop
% \WsPaper         % uncomment for final version of EG Workshop contribution
%
 \electronicVersion % can be used both for the printed and electronic version

% !! *please* don't change anything above
% !! unless you REALLY know what you are doing
% ------------------------------------------------------------------------

% for including postscript figures
% mind: package option 'draft' will replace PS figure by a filname within a frame
\ifpdf \usepackage[pdftex]{graphicx} \pdfcompresslevel=9
\else \usepackage[dvips]{graphicx} \fi

\PrintedOrElectronic

% prepare for electronic version of your document
\usepackage{t1enc,dfadobe}

\usepackage{egweblnk}
\usepackage{cite}
\usepackage[color=yellow!30]{todonotes}
\usepackage{color,soul}

% For backwards compatibility to old LaTeX type font selection.
% Uncomment if your document adheres to LaTeX2e recommendations.
% \let\rm=\rmfamily    \let\sf=\sffamily    \let\tt=\ttfamily
% \let\it=\itshape     \let\sl=\slshape     \let\sc=\scshape
% \let\bf=\bfseries

% end of prologue

% \input{eurovis_guidelines_body.tex}
% \input{pathways_taxonomy_body.tex}

% ---------------------------------------------------------------------
% EG author guidelines plus sample file for EG publication using LaTeX2e input
% D.Fellner, v1.17, Sep 23, 2010


\title[A Task Taxonomy to support Visualization for the Effective Analysis of Biological Pathways]%
      {A Task Taxonomy to support Visualization for the Effective Analysis of Biological Pathways}

% for anonymous conference submission please enter your SUBMISSION ID
% instead of the author's name (and leave the affiliation blank) !!
\author[]{SUBMISSION ID}

% ------------------------------------------------------------------------

% if the Editors-in-Chief have given you the data, you may uncomment
% the following five lines and insert it here
%
% \volume{27}   % the volume in which the issue will be published;
% \issue{1}     % the issue number of the publication
% \pStartPage{1}      % set starting page


%-------------------------------------------------------------------------
\begin{document}

% \teaser{
%  \includegraphics[width=\linewidth]{eg_new}
%  \centering
%   \caption{New EG Logo}
% \label{fig:teaser}
% }

\maketitle

\begin{abstract}
Understanding complicated networks of interactions and chemical components is essential to solving contemporary problems in modern biology, especially in domains such as cancer and systems research.
In these domains, biological pathway data is used to represent chains of interactions that occur within a given biological process.
Visual representations can help researchers understand and interact with complex pathway data in a number of ways.
Biological data sets offer unique challenges for visualization, due to their complexity and heterogeneity.

Here, we present taxonomy of tasks -- generated from interviews with several domain experts -- that are regularly performed by researchers who work with biological pathway data.
These tasks require further classification this is provided by existing taxonomies.

\todo[inline]{more here about how these tasks are different from previous work}

We also examine the existing visualization techniques which support each of the tasks, and we discuss gaps in the existing visualization space revealed by our taxonomy.
We conclude by suggesting future research directions based on our taxonomy and motivated by the comments received by our domain experts.


\begin{classification} % according to http://www.acm.org/class/1998/
\CCScat{Computer Graphics}{I.3.3}{Picture/Image Generation}{Line and curve generation}
\end{classification}

\end{abstract}

%-------------------------------------------------------------------------
\section{Introduction}

Understanding complicated networks of bio-molecular interactions and chemical components is essential to solving contemporary problems in modern biology, especially in domains such as cancer and systems research~\cite{hanahan2011hallmarks}.
These bio-molecular interactions are represented in the form \emph{pathways}, which are used to describe a chain of interactions between biochemical and biological entities within a cell.
Pathways are small, curated subsets of a much larger, complex graph of interactions between molecules, and a given pathway usually represents a particular biological process that is relevant within some research context.
% In order to limit the scope of their analyses...
For example, Figure \ref{fig:kvik} shows a typical representation of a pathway as a human-curated node-link diagram, where nodes are biological entities and edges represent interactions between them.

\begin{figure}[htb]
  \centering
  \includegraphics[width=\linewidth]{figures/kegg2}
  \caption{\label{fig:kvik} A view of a typical KEGG diagram. From~\cite{Fjukstad2014kvik}.}
\end{figure}

%\p{Pathways are very complex}

Researchers who work with pathway data are confronted with a number of challenges.
Pathway files may contain hundreds of proteins and biomolecules that participate in a variety of reactions.
In an abstract sense, reactions can be seen as state transitions with multiple inputs and outputs.
Participants --- genes, proteins, and other molecules within a cell --- can act as inputs or outputs to multiple reactions, and the relationships between reactions inherently include feedback loops.
Reactions often have an effect on other reactions, inhibiting or promoting their frequency.
These molecular activation pathways are inherently dynamic, which limits the utility of any static graph representation \cite{kitano2002systems}.
Representing complexity while also enabling researchers to see higher order patterns is a significant challenge \cite{saraiya2005visualizing}.

%\p{Pathways are useful for presentation}

Pathway diagrams can be useful for both presentation and analysis.
For presentation, pathway diagrams can contextualize a set of biological processes within a cell, and diagrams often show the location of cellular membranes in order to help provide a frame of reference for a given process.
Ideally, a pathway diagram allows a viewer to efficiently understand a complex set of biological relationships.

%\p{Pathways are useful for analysis}

While pathways may be useful for presenting and contextualizing a set of reactions, they can also be an important part of analyses in domains related to molecular biology and systems research, among others.
%\p{Domain context}

For example, molecular activation pathways are of critical importance to cancer researchers, who hope to understand --- and potentially disrupt --- malignant cycles of uncontrolled cellular growth, replication, and mediated cell death \cite{cairns2011regulation}.
Effective cancer drug development involves determining how proteins affected by a drug in turn affect important cellular pathways, and in this domain the downstream consequences of a particular drug effect are especially important \cite{luo2003targeting}.
In a separate domain, stem-cell researchers work with pathways that will precipitate a desired cellular differentiation into specific cell types \cite{reya2001stem}.

%\p{Static representations are not enough}

In the last decade, analyses that involve hundreds or thousands of genes and gene products have become common. When analyzing such large and complex data, visual representations can be essential.
Often, static representations are inadequate.
The complexity and amount of information that needs to be incorporated in given diagram can make static representations cluttered and difficult to interpret.
Thus, modern tools make careful use of user interactions and visualization techniques to allow a user to effectively explore and analyze pathway data.

%\p{Designing effective tools}

Designing effective visual analytics applications requires a detailed understanding of analysis tasks that are performed by the user.
Pathway data are often large and complex, and analysts will want to perform a variety of tasks depending on their research domain.
Tasks may be exploratory in nature, and a useful visualization of pathway data could reveal new insights to a researcher.
Tasks may also involve detailed queries or calculations of various network metrics, for example.
A comprehensive understanding of tasks performed by domain researchers in a typical analysis is essential to the design and implementation of an effective visual analytics application.

%\p{Other reviews}

%Here we perform a comprehensive analysis of tasks and requirements in an effort to design effective platforms for visual analytics of pathway data.
%Previous reviews of pathway analysis tools~\cite{Gehlenborg2010omics,Suderman2007tools} have surveyed the population of available applications.
%However, the most recent review was published over five years ago, and it includes only a surface-level discussion of tasks, requirements, and visual encoding techniques.

%\p{In this work}

In this work, we present a description and analysis of tasks and requirements related to biological pathway research.
Tasks were gathered from several interviews with domain experts who work with biological pathway data.
After an introduction to the structure and content of pathway data, we describe the tasks that were garnered from our interviews.
%Using these tasks, we then describe the high-level requirements of an effective visual analytics platform for pathway data.
We then review visual representations of pathway data in the context of our requirements.
We also review existing tools that implement those visual representations.
Finally, avenues of future research are considered, along with a brief summary of lessons learned from domain experts.

\subsection{Pathway data}

In order to aid an understanding of pathway visualization tools, an understanding of the structure of pathway data structures is necessary.
In this section we briefly explain the structure of typical pathway data files.

\subsubsection{Pathway Data Model}

Information stored in any pathway data file can generally be broken down into three components:

\begin{itemize}

\item \textbf{Entity}\\
An entity is a component of a pathway such as a gene, a gene product (i.e. a protein), a complex of proteins, or a small biomolecule within a cell.
Entities are identified by name and are involved in one or more relationships.
Importantly, pathways themselves can be entities within other pathways.
\item \textbf{Relationship}\\
A relationship involves two or more entities.
Various kinds of relationships which different biological meanings are present in a pathways.
Relationships can be directed or undirected, and they can involve more than two entities (meaning the resulting network is a hyper-graph).
\item \textbf{Meta-data}\\
The complex nature of the information stored in a pathway requires additional data to be stored with each entity and relationship.
Meta-data can include experimental data, scientific information such as the molecular structure of a chemical compound, as well as links to additional resources or publications related to an entity or relationship.

\end{itemize}


\subsubsection{Pathway Data Formats}

Pathway data can be stored in one of several file formats.
In particular, \textit{BioPAX}~\cite{demir2010biopax}, \textit{KEGG}~\cite{kanehisa2000kegg} and \textit{SBML} \cite{Hucka2003} are the most popular standards for storing the complex data structures described in the previous section.

These formats are XML-based and represent data as an ontology.
\emph{BioPAX}, in particular, was designed to be a general format for biological pathways across a variety of domain contexts~\cite{demir2010biopax}.
Systems Biology Graph Notation~\cite{Novere2009} is a visual standard often used to visualize \textit{BioPAX} and \textit{SBML} file formats.
Other formats are employed for the visualization of biological pathways that are not specific to the field of biology.
For instance \textit{SIF Simple Interaction Format} used by \textit{Cytoscape}~\cite{Shannon2003cytoscape} is used to visualize undirected interactions between participants.
%-------------------------------------------------------------------------
\section{Related Work}
%-------------------------------------------------------------------------
\subsection{Biological Pathway Visualization}
\todo[inline]{Need more here on history and importance}
Biological networks are an important application domain and visualizations are an important tool to understand complex biological processes.
%High throughput techniques have resulted in experiments producing vast amounts of highly complex data.

There are a large number of tools available and many existing surveys describe them \cite{Suderman2007tools,pavlopoulos2008survey,Gehlenborg2010omics}.
In this paper we highlight examples of existing tools and techniques which provide strong support for the tasks described in our taxonomy, however it is not intended as a complete survey of biological visualization applications.

%However, the most recent review was published over five years ago, and it includes only a surface-level discussion of tasks, requirements, and visual encoding techniques.



%-------------------------------------------------------------------------
\subsection{Task Taxonomies}
The field of visual analytics has produced a number of \textit{task taxonomies}, which are written in an effort to understand exactly how various analytics tasks are enabled by different visualization techniques, and vice versa. These taxonomies help clarify the utility of existing techniques while also providing a low-level template for the design and evaluation of new techniques.
Wehrend and Lewis~\cite{Wehrend1990} provide one of the earliest visualization task taxonomies, with the goal of ``accelerating progress in scientific visualization'' by allowing researchers to easily find the right visualization technique for a given problem.
Schneiderman~\cite{Shneiderman1996} defined a ``task by data type taxonomy'' for in formation visualization in order to \textit{``to sort out the prototypes and guide researchers to new opportunities''}.
These seminal taxonomies were, like many later taxonomies, independent of a specific visualization application domain --
their purpose was to provide a low-level description and categorization of the analysis tasks enabled by \textit{any} visualization of data.

Later taxonomies focused specifically on more narrow categories of visualization.
For instance, Valiati et al.~\cite{Valiati2006} provide a taxonomy focused specifically on multidimensional visualizations. They build on~\cite{Wehrend1990}, but focus on tasks uniquely related to multidimensional visualizations (such as parallel coordinates).
Like previous authors, their goal is to guide the choices of visualization and interaction techniques, and also to  help support usability testing.
Lee at al~\cite{Lee2006} define a graph visualization taxonomy of tasks that are frequently encountered when analyzing graph data.
The stated goal of this work was to improve the evaluation of graph visualization systems by creating a set of common benchmark tasks (which could be used in conjunction with benchmark data sets).
Their taxonomy covers tasks for the analysis of graphs in general, and was inspired by example tasks from many domains that make regular use of graph data.
The authors built on Amar and Stasko's~\cite{Amar2005} low level visual analytic task list by composing existing low-level tasks into higher level complex tasks while also proposing additional tasks that are not captured by low-level tasks presented in existing taxonomies.

\todo[inline]{Consistent hypenation: ``low-level'' or ``low level''? }

\todo[inline]{When writing about previous work, need to be consistent on present vs. past tense. Which to use?}

Several recent taxonomies have focused on aspects of graph visualization that extend the work of Lee at al~\cite{Lee2006}.
Ahn et al~\cite{Ahn2014} provide a task taxonomy for analyzing networks that evolve over time, also known as dynamic graphs.
The dynamic and complex nature of dynamic graph data yields a similarly complex set of analysis tasks, and many of these tasks are not covered by the general graph taxonomy of Lee at al~\cite{Lee2006} -- thus, new tasks need to be specified.
Pretorius et al~\cite{Pretorius2014} focus on multivariate graph visualization (where graph elements contain multiple attributes).
Their work builds on the work of both Lee at al~\cite{Lee2006} and of Valiati et al.~\cite{Valiati2006}, as multivariate networks can be considered a multidimensional data set.

The earliest visualization taxonomies were written as very general classifications of low-level analytic tasks related to any data visualization. In more recent publications, and as visualization research has progressed, task taxonomies have increasingly focused on more constrained subsets of tasks related to particular types of data structures.

While recently-published task taxonomies have focused on particular data structures (or for datasets with particular characteristics), there have been few \hl{or none?} written in the context of a particular application domain.

\todo[inline]{One could argue that the whole point of a taxonomy is to be general, and to ignore domain. We should highlight the benefits of writing a taxonomy based on domain-specific tasks. Or how this interesting... etc.}

Biological Pathway visualization is a complex application domain.
Previous taxonomies have avoided being domain specific, however this domain poses many specific challenges not encountered in the created of the previous taxonomies
The underlying data sets are dynamic multivariate hyper-graphs, and are more complex than any of those described in previous taxonomies.
The tasks to be completed by biologists are also highly complex, involving many different entity and relationship types, and are not fully covered by the existing taxonomies.
\todo[inline]{Need to be 100 percent sure this is true}
%Amar and stasko do include correlation as one of their tasks
% pretorius also includes causation...

%-------------------------------------------------------------------------

\section{Interviews}

Interviews were conducted with seven biological scientists, each of whom works with pathway data in some form.
Those interviewed included one tenured professor, three assistant professors, one researcher at a cancer research institution, one postdoctoral research associate, and one masters student in bioinformatics.
Interviews were loosely structured, but interview questions were designed to elicit a detailed understanding of the tasks performed by the researcher in a typical analysis, as well as an understanding of the type and structure of data that each researcher worked with.
Each researcher also presented their views on the utility of pathway data and pathway diagrams in general.

%-------------------------------------------------------------------------
\section{Task Taxonomy}

\subsection{Taxonomy Overview}

Biological pathways are generally represented as weighted, directed graphs and in some cases include hyper-edges and compound nodes.
Thus, the low-level identification of nodes, vertices, and their attributes is essential.
Detailed task taxonomies have been created which describe tasks related to particular types of data, including multivariate graphs.
In particular...
\hl{importance of understanding relationships between compound nodes}

\subsection{Attribute Tasks}
\subsubsection{Task: Identify Vertex Attributes}
\subsubsection{Task: Identify Edge Attributes}
\subsubsection{Task: Uncertainty and Provenance}
Several of the researchers mentioned certain tasks related to the curation, maintenance, and understanding of pathway data. RG mentioned the importance of being able to \emph{debug} potentially flawed data. NH and FZ both expressed a need to create ``personalized'' pathways that only include a user-determined subset of entities and relationships. Finally, QW and FZ discussed the importance of understanding \emph{uncertainty} within pathway data. For example, each relationship within a BioPAX file is usually associated with a publication that provides evidence for its existence. Thus, some relationships may be subject to scrutiny within the scientific community, while others may have more robust empirical support.

\subsection{Relationship Tasks}

Understanding how pathway entities are connected was of critical importance to all of the researchers we interviewed, and is essential to most research in bioinformatics.

While some analyses and datasets involve undirected relationships between genes or gene products, studies of metabolic networks and other inter-cellular processes rely on directed relationships, and several researchers that we interviewed stressed the importance of understanding directed relationships between entities.

\hl{merge with hierarchical tasks}

Pathway data is also inherently hierarchical.
These hierarchical relationships describe relationships of containment that are either abstract or based on real biochemical interactions within a cell.
For example, a pathway (itself an abstraction) can be nested within other pathways.
These nested pathways generally encapsulate some commonly-understood hierarchy of biological processes that take place within a cell, such as cellular replication.
Other representations include the more general notion of a ``module'' of connected components, such as gene products.
Hierarchical relationships can also represent physical interactions between biochemical participants.
A common of example of this is in bio-molecular complexes, which are themselves composed of other complexes or biomolecules.

While in a strict sense, a hierarchical relationship between nodes can be seen as one particular type of edge, we instead explicitly distinguish between between hierarchical and non-hierarchical relationships.
This distinction is motivated by the observation that in most cases, pathway data includes relationships of hierarchy (i.e., when one vertex is contained within another) \textit{in parallel} with other, non-hierarchial relationships, such as the relationship between one gene product that activates or inhibits another.

A vertex that contains other entities can be represented as a \texit{compound node}, which is equivalent to a ``parent'' vertex or in some contexts a ``module.''

Also, note that while non-hierarchical relationships can take a variety of forms, the only form of hierarchical relationship is one of \texit{containment}, from parent to child, and is undirected.

\todo[inline]{We need to be consistent about how we talk about compound nodes aka parent nodes aka modules}

\subsection{Relationship Tasks}

identification of:
\\
\\
relationship type: adjacency, hierarchy
\\ $\times$
node type: simple, compound
\\ $\times$
from: one, many
\\ $\times$
to: one, many
\\ $\times$
directed, undirected

% \subsubsection{Task: Identify Adjacency, one-to-many}
% Given an entity, find the entities that it is connected to.
%
% \subsubsection{Task: Identify Adjacency, many-to-many}
% Given a set of entities, find all other entities that are connected to that set.
%
% \subsubsection{Task: Identify Adjacency, one-to-many, outgoing}
% Given an entity, find the entities that it is connected to via outgoing edges.
%
% \subsubsection{Task: Identify Adjacency, one-to-many, incoming}
% Given an entity, find the entities that it is connected to via incoming edges.

\subsubsection{Task: Identify Containment, parent-to-parent}
\subsubsection{Task: Identify Containment, parent-to-leaf}

\subsection{Compositions of Adjacency Tasks and Hierarchy Tasks}
% Given our distinction between hierarchical and non-hierarchical relationships, we can characterize compositions of tasks related to the identification of relationships between entities and hierarchies.
%
% These relationships occur frequently in biological pathway data.
%
% \todo[inline]{examples}

It is important to note that a one-to-one relationship between an entity and a parent is \textit{not} the same as a one-to-many relationship between an entity and all of that parents children.
For instance, BioPax data contains the abstract ``NextStep'' relationship, which defines, as the name suggests, an arbitrary notion of the ``next step'' of some biological process.
A biochemical reaction could be connected, via a \textit{single} ``NextStep'' relationship, to an entire pathway, which could potentially contain thousands of nodes.
This relationship is clearly not the same as a biochemical reaction being connected to every entity within a pathway.

\subsubsection{Task: Identify Adjacency, parent-to-parent, one-to-one}
\subsubsection{Task: Identify Adjacency, entity-to-parent, one-to-one}
\subsubsection{Task: Identify Adjacency, parent-to-parent, one-to-many}
\subsubsection{Task: Identify Adjacency, entity-to-parent, one-to-many}
\todo[inline]{Should we further sub-divide into directed and undirected?}

\subsection{Cause and Effect Tasks}
A category of tasks inherent to a variety of work in bioinformatics is the identification of \textit{causal relationships} that exist between bio-molecular entities.
When discussing directed paths between entities, one entity is said to be \emph{upstream} or \emph{downstream} of another.

Understanding these upstream and downstream relationships is particularly important to domains such as cancer drug research, where a drug may affect a small subset of genes or gene products, which in turn will affect various downstream processes.
\emph{Causal networks} are also particularly essential to the analysis of large-scale gene expression data.
For instance, a causal network could reveal the likely regulators of a set of genes that are observed to be up-regulated or down-regulated in a particular setting~\cite{felciano2013predictive, Kramer2013ipa-causal}.

In most cases, a directed relationship is meant to represent a biochemical reaction, where one entity is consumed as a reactant and another is produced as a product.
Thus, an upstream entity may be connected to a downstream entity through a chain of several directed links.
In the most basic sense, the ``entities'' mentioned above are genes, gene products (such as proteins or complexes), or other small molecules within a cell.
A researcher may be interested in understanding the path of reactions (or other relationships) that connects two entities.

\todo[inline]{More on Activation and Inhibition? Activation and inhibition vs up-regulation and down-regulation?}

\subsubsection{Task: Identify Cascading Effects, one-to-one}
\subsubsection{Task: Identify Cascading Effects, one-to-many}
\subsubsection{Task: Identify Cascading Effects, many-to-many}
\subsubsection{Task: Identify Cascading Effects, many-to-one}


\subsubsection{Task: Identify Feedback Loops}

\subsection{Compound Relationships and Multiple Datasets}
\todo[inline]{todo}

\subsection{Pathway Modification and Curation}
\todo[inline]{todo}
\todo[inline]{curating, annotating, debugging}



%-------------------------------------------------------------------------
\section{Discussion}

\subsection{Future Research Directions}

\subsection{Visualizing Uncertainty}

Especially considering our feedback from domain experts, tools generally do not attempt to visualize the ``uncertainty'' behind a connection in a pathway, as expressed by the first domain expert. This is a challenging task, as even the definition of ``uncertainty'' may be difficult to operationalize. However, data formats such as \emph{BioPAX} do have robust support for citations, allowing published references to be connected to entities and relationships within a pathway. A tool that could effectively encode ``uncertainty data'' into a visualization may be very valuable to systems researchers who work with the results of hundreds or thousands of separate publications.

\section{Conclusions}



%-------------------------------------------------------------------------

%\bibliographystyle{eg-alpha}
\bibliographystyle{eg-alpha-doi}

\bibliography{references}


\end{document}
